.github/workflows/rpc-health-nightly.yml

```diff
@@ -0,0 +1,48 @@
+name: RPC Health Nightly Checks
+
+on:
+  schedule:
+    # Run at 00:00 UTC every day
+    - cron: '0 0 * * *'
+  # Allow manual triggering
+  workflow_dispatch:
+
+jobs:
+  install:
+    uses: ./.github/workflows/reusable-install-build.yml
+    with:
+      job: install
+
+  build:
+    uses: ./.github/workflows/reusable-install-build.yml
+    with:
+      job: build
+    needs: [install]
+
+  rpc-health-mainnet:
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            **/node_modules
+            .yarn/cache
+          key: ${{ runner.os }}-yarn-cache-${{ hashFiles('./yarn.lock') }}
+      - name: test
+        run: yarn run test:rpc-health-mainnet
+
+  rpc-health-testnet:
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            **/node_modules
+            .yarn/cache
+          key: ${{ runner.os }}-yarn-cache-${{ hashFiles('./yarn.lock') }}
```

Comment:
if you keep the reusable action thing, you should make the rpc health tasks rely on install, and then reuse the install-build thing as a step within the rpc-health ones. that way you can reuse the checkout/cache/recovery of the reusable action and the rpc test jobs just become 
- reusable build
- test